### PR TITLE
Chore/add types to style handler

### PR DESCRIPTION
**Pull request by saltcod**
## What kind of change does this PR introduce?

Refactor of existing code

## What is the current behavior?

The `styleHandler` function that is used to access the `ThemeContext` is really a hook and should be named as such. It also lacked typing, so you had to dig into the `defaultTheme.ts` file to know which style objects/variants were defined. Luckily only a few places were accessing variants that didn't really exist, and I think those end up getting thrown out when the class names are joined together anyway. Still - better to know what the object shape you're working with.

## What is the new behavior?

`styleHandler` has been renamed to `useStyles` and returns a type of `DefaultTheme[key]`. I also added types to `RadioContext`.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by saltcod** [comment]
cc @brandongregoryscott all still look good to you on this one?

File: packages/ui/src/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -4,7 +4,8 @@ import { IconAlertTriangle } from '../Icon/icons/IconAlertTriangle'
 import { IconInfo } from '../Icon/icons/IconInfo'
 import { IconX } from '../Icon/icons/IconX'
 
-import styleHandler from '../../lib/theme/styleHandler'
+// @ts-ignore
+import useStyles from '../../lib/theme/use-styles'
```
Comment (brandongregoryscott) [inline packages/ui/src/components/Alert/Alert.tsx:8]: @saltcod Is this `@ts-ignore` necessary? Might've been from the merge conflict resolution, but this might clobber the types for the hook

File: packages/ui/src/lib/constants.ts [hunk 2]
```diff
@@ -11,9 +11,7 @@ export const COLORS = [
   'indigo',
   'blue',
   'green',
-  'grass',
```
Comment (brandongregoryscott) [inline packages/ui/src/lib/constants.ts:14]: @saltcod Were these changes intentional? I don't remember editing this file

**Review by brandongregoryscott — APPROVED** [review]
Left two comments around changes I saw from my original PR, but looks good to me! Thanks for taking a look. @MildTomato might have some idea around the `@ts-ignore` comments in the [`Tabs`](https://github.com/supabase/supabase/pull/13813/files#diff-c2ad1dbf86e044ac8f970484f547c497efae85ae0b7fd44dac56f4f889572f75) component (`cards` variant doesn't exist based on what is currently in the theme object)

File: packages/ui/src/lib/theme/defaultTheme.ts [hunk 3]
```diff
@@ -227,7 +227,6 @@ export default {
       indigo: `bg-indigo-200 text-indigo-1100 border border-indigo-700`,
       blue: `bg-blue-200 text-blue-1100 border border-blue-700`,
       green: `bg-green-200 text-green-1100 border border-green-700`,
-      grass: `bg-grass-200 text-grass-1100 border border-grass-700`,
```
Comment (brandongregoryscott) [inline packages/ui/src/lib/theme/defaultTheme.ts:230]: I don't remember making this change either - if this is related to some of the Radix color cleanup that I saw recently merged in, disregard 👍

Comment (saltcod) [reply packages/ui/src/lib/theme/defaultTheme.ts:230]: Yup that's me from earlier

**Comment by saltcod** [comment]
@brandongregoryscott any chance you could look at the conflicts here? Love to get this in this week before it goes any longer — apologies for the delay already...

**Comment by brandongregoryscott** [comment]
@saltcod I pushed up the merge conflict resolutions to my original branch: https://github.com/brandongregoryscott/supabase/tree/chore/add-types-to-style-handler

**Comment by MildTomato** [comment]
hey @saltcod @brandongregoryscott 

I'm working on moving components to CVA and so perhaps some of this work will become redundant, as the style handler will get dropped in the process.

Didn't want you folks getting buried in work that might get replaced.
